apps/search/js/search.js

```diff
@@ -246,6 +261,16 @@
     },
 
     /**
+     * Called to notify the AppUsageMetrics of a search
+     */
+    metricsNotify: function(provider) {
+      this._portMetrics.postMessage({
```

Comment:
My understanding is that the Rocketbar in the system app will load the Search app.  When the Rocketbar gets put into the active state, it loads the SearchApp at [1].  Then it waits for that to be loaded via a promise (I think this goes back to the initial problem we had with putting this in rocketbar on |make reset-gaia|.  I believe this then triggers the init function which initializes the ports at [2].

[1] https://github.com/tamarahills/gaia/blob/master/apps/system/js/rocketbar.js#L124
[2] https://github.com/tamarahills/gaia/blob/master/apps/search/js/search.js#L367